CS/XPO/Readme.md

```diff
@@ -0,0 +1,103 @@
+
+2. For apps with no security or with the Client-Side Security (`XPObjectSpaceProvider` or `SecuredObjectSpaceProvider`):
+   
+   In the _YourSolutionName.Wxx/WxxApplication.xx_ files, modify the `CreateDefaultObjectSpaceProvider` method to call the `SequenceGenerator.Initialize` method as shown in the _Demo.Wxx\WxxApplication.xx_ files:
+   
+   ```cs
+   protected override void CreateDefaultObjectSpaceProvider(CreateCustomObjectSpaceProviderEventArgs args) {
+       IXpoDataStoreProvider dataStoreProvider = XPObjectSpaceProvider.GetDataStoreProvider(args.ConnectionString, args.Connection, true);
+       // Web:
+       // IXpoDataStoreProvider dataStoreProvider = GetDataStoreProvider(args.ConnectionString, args.Connection);
+       GenerateUserFriendlyId.Module.SequenceGenerator.Initialize(dataStoreProvider);
+       ...
+   }
+   
+   ```
+   
+   For apps with the Middle Tier Security (`DataServerObjectSpaceProvider`):
+   
+   In the _YourSolutionName.ApplicationServer_ project, locate and modify the `serverApplication_CreateCustomObjectSpaceProvider` or `CreateDefaultObjectSpaceProvider`  methods to call the `SequenceGenerator.Initialize` method in the same manner.
+
+3. If you are using pure XPO classes, then inherit your business classes to which you want to add sequential numbers from the module's `UserFriendlyIdPersistentObject` class. Declare a calculated property that uses the `SequenceNumber` property of the base class to produce an string identifier according to the required format:
+   
+   ```cs
+   public class Contact : GenerateUserFriendlyId.Module.BusinessObjects.UserFriendlyIdPersistentObject {
+       [PersistentAlias("Concat('C',PadLeft(ToStr(SequentialNumber),6,'0'))")]
+       public string ContactId {
+           get { return Convert.ToString(EvaluateAlias("ContactId")); }
+       }
+   
+   ```
+   
+   If you are using DC interfaces, then implement the `IUserFriendlyIdDomainComponent` interface by your custom domain component:
+   
+   ```cs
+   public interface IDocument : GenerateUserFriendlyId.Module.BusinessObjects.IUserFriendlyIdDomainComponent {
+       [Calculated("Concat('D',PadLeft(ToStr(SequentialNumber),6,'0'))")]
+       string DocumentId { get; }
+   
+   ```
+   
+   Additionally for DC, use `UserFriendlyIdPersistentObject` as a base class during your custom domain component registration, for example:
+   
+   ```cs
+   XafTypesInfo.Instance.RegisterEntity("Document", typeof(IDocument), typeof(GenerateUserFriendlyId.Module.BusinessObjects.UserFriendlyIdPersistentObject));
+   
+   ```
+   
+   Note, that the sequential number functionality shown in this example does not work with [DC shared parts](http://documentation.devexpress.com/#Xaf/DevExpressExpressAppDCITypesInfo_RegisterSharedParttopic), because it requires a custom base class, which is not allowed for shared parts.
```

Comment:
```suggestion
   Note that the sequential number functionality shown in this example does not work with [DC shared parts](http://documentation.devexpress.com/#Xaf/DevExpressExpressAppDCITypesInfo_RegisterSharedParttopic), because it requires a custom base class, which is not allowed for shared parts.
```